gapic/schema/wrappers.py

```diff
@@ -741,6 +742,41 @@ def field_headers(self) -> Sequence[str]:
 
         return next((tuple(pattern.findall(verb)) for verb in potential_verbs if verb), ())
 
+    @property
+    def http_options(self) -> List[Dict[str, str]]:
+        """Return a list of the http options for this method.
+
+        e.g. [{'method': 'post'
+              'uri': '/some/path'
+              'body': '*'},]
+
+        """
+        http = self.options.Extensions[annotations_pb2.http]
+        http_options = copy.deepcopy(http.additional_bindings)
+        http_options.append(http)
+        answers : List[Dict[str, str]] = []
+
+        for http_rule in http_options:
+            try:
+                method, uri = next((method, uri) for method,uri in [
+                    ('get',http_rule.get),
+                    ('put',http_rule.put),
+                    ('post',http_rule.post),
+                    ('delete',http_rule.delete),
+                    ('patch',http_rule.patch),
+                    ('custom.path',http_rule.custom.path),
+                    ] if uri
+                )
+            except StopIteration:
+                continue
```

Comment:
I think it's better not to catch a `StopIteration` and instead use explicit control flow.
This is a fairly common pattern when using `next`:
```python
thing = next((generator_expression_or_iterable), None) # Or (None, None) if detupling
if not thing:
    continue # Or do some other action
```